server/src/main/java/org/elasticsearch/plugins/PluginsService.java

```diff
@@ -293,6 +294,33 @@ static void loadExtensions(Collection<LoadedPlugin> plugins) {
         }
     }
 
+    // package private for testing
+    <T> Iterator<T> providersIterator(Class<T> service, ClassLoader classLoader) {
```

Comment:
I think this can be folded into loadServiceProviders now that the tests don't mock it?